compose/service.py

```diff
@@ -780,14 +780,22 @@ def build_container_name(project, service, number, one_off=False):
 
 # Images
 
-
 def parse_repository_tag(s):
-    if ":" not in s:
-        return s, ""
-    repo, tag = s.rsplit(":", 1)
-    if "/" in tag:
-        return s, ""
-    return repo, tag
+    tag_sep = ":"
+    digest_sep = "@"
+
+    repo, tag, separator = s, "", tag_sep
+
+    if digest_sep in s:
+        separator = digest_sep
+        repo, tag = s.rsplit(digest_sep, 1)
+
+    elif tag_sep in s:
+        repo, tag = s.rsplit(tag_sep, 1)
+        if "/" in tag:
+            repo, tag = s, ""
+
+    return repo, tag, separator
```

Comment:
Doc string is definitely worth adding.
I like a explicit and easy to understand variable names as well as early returns.
However 4 returns (3 in the row) in 16 lines of code is a bit to much. How do you think? @mnowster